TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs

```diff
@@ -104,7 +105,8 @@ public abstract class SerializableUIOptionBase<TVal, TUI, TComponent> : ILegacyS
 
         /* UI: */
 
-        public bool HasUI => _ui != null;
+        // TODO FIX - temporary solution!
+        public bool HasUI => _ui != null && Thread.CurrentThread != SimulationManager.instance.m_simulationThread;
```

Comment:
can you make sure its in main thread?

Reply:
How to get main thread handle?

Reply:
```cs
bool InMainThread() => Dispatcher.currentSafe == ThreadHelper.dispatcher;
```

Reply:
now that I read the code again I am not sure if that really is the main thread.

Reply:
hmm, I'm pretty sure it is, looking in the code, it's just not so obvious reading code of your proposed helper